firefox-ios/Client/Telemetry/GleanWrapper.swift

```diff
@@ -7,21 +7,103 @@ import Glean
 
 protocol GleanWrapper {
     func handleDeeplinkUrl(url: URL)
-    func submitPing()
     func setUpload(isEnabled: Bool)
+    func submitPing()
+
+    // MARK: Glean Metrics
+
+    func submitEventMetricType<ExtraObject>(event: EventMetricType<ExtraObject>,
+                                            extras: EventExtras) where ExtraObject: EventExtras
+    func submitEventMetricType<NoExtras>(event: EventMetricType<NoExtras>) where NoExtras: EventExtras
+    func submitCounterMetricType(event: CounterMetricType)
+    func submitStringMetricType(event: StringMetricType, value: String)
+    func submitLabeledMetricType(event: LabeledMetricType<CounterMetricType>, value: String)
+    func submitBooleanMetricType(event: BooleanMetricType, value: Bool)
+    func submitQuantityMetricType(event: QuantityMetricType, value: Int64)
+
+    func addToNumeratorRateMetricType(event: RateMetricType, amount: Int32)
+    func addToDenominatorRateMetricType(event: RateMetricType, amount: Int32)
+
+    func startMeasurementTelemetry(forMetric metric: TimingDistributionMetricType) -> GleanTimerId
+    func cancelMeasurementTelemetry(forMetric metric: TimingDistributionMetricType,
+                                    timerId: GleanTimerId)
+    func stopAndAccumulate(forMetric metric: TimingDistributionMetricType,
+                           timerId: GleanTimerId)
 }
 
 /// Glean wrapper to abstract Glean from our application
 struct DefaultGleanWrapper: GleanWrapper {
-    public static let shared = DefaultGleanWrapper()
-
     func handleDeeplinkUrl(url: URL) {
         Glean.shared.handleCustomUrl(url: url)
```

Comment:
are we planning to create tickets to updated these as well since they are relying on `Glean.shared`?

Reply:
`Glean.shared` is the framework itself, I am not sure what we could do there to change this apart from having the wrapper on top of it. Do you have any suggestions?

Reply:
hmm what about injecting it? not a blocker though, can think more when return from the holidays